File: src/creational/factorypattern/example2/factory/VehicleFactory.java
package creational.factorypattern.example2.factory;

public class VehicleFactory {
    public static Vehicle getVehicle(String vehicleType){
        Vehicle vehicle;
        if (vehicleType.equals("car")) {
            vehicle = new Car();
        } else if (vehicleType.equals("bike")) {
            vehicle = new Bike();
        } else {
            vehicle = null;
        }
        return vehicle;
    }
}

File: Creational/Factory/vehicle/vehicle_factory.cpp
#include "vehicle_factory.hpp"

Vehicle* VehicleFactory::getVehicle(const std::string& vehicleType)
{
    Vehicle* Vehicle = nullptr;
    if(vehicleType=="car")
    {
        Vehicle = new Car();
    }
    else if(vehicleType == "bike")
    {
        Vehicle = new Bike();
    }
    return Vehicle;
}